habitat-baselines/habitat_baselines/config/default_structured_configs.py

```diff
@@ -0,0 +1,366 @@
+    height: int = 256
+    width: int = 512
+    sensor_uuids: List[str] = field(default_factory=lambda: [
+        "BACK",
+        "DOWN",
+        "FRONT",
+        "LEFT",
+        "RIGHT",
+        "UP",
+    ])
+
+
+@dataclass
+class Cube2FishConfig(ObsTransformConfig):
+    height: int = 256
+    width: int = 256
+    fov: int = 180
+    params: Tuple[float] = (0.2, 0.2, 0.2)
+    sensor_uuids: List[str] = field(default_factory=lambda: [
+        "BACK",
+        "DOWN",
+        "FRONT",
+        "LEFT",
+        "RIGHT",
+        "UP",
+    ])
+
+
+@dataclass
+class Eq2CubeConfig(ObsTransformConfig):
+    height: int = 256
+    width: int = 256
+    sensor_uuids: List[str] = field(default_factory=lambda: [
+        "BACK",
+        "DOWN",
+        "FRONT",
+        "LEFT",
+        "RIGHT",
+        "UP",
+    ])
+
+
+@dataclass
+class PolicyConfig(HabitatBaselinesBaseConfig):
+    name: str = "PointNavResNetPolicy"
+    action_distribution_type: str = "categorical"  # or 'gaussian'
+    # If the list is empty, all keys will be included.
+    # For gaussian action distribution:
+    action_dist: ActionDistributionConfig = ActionDistributionConfig()
+    obs_transforms: Any = MISSING
+
+
+@dataclass
+class PPOConfig(HabitatBaselinesBaseConfig):
+    """Proximal policy optimization config"""
+    clip_param: float = 0.2
+    ppo_epoch: int = 4
+    num_mini_batch: int = 2
+    value_loss_coef: float = 0.5
+    entropy_coef: float = 0.01
+    lr: float = 2.5e-4
+    eps: float = 1e-5
+    max_grad_norm: float = 0.5
+    num_steps: int = 5
+    use_gae: bool = True
+    use_linear_lr_decay: bool = False
+    use_linear_clip_decay: bool = False
+    gamma: float = 0.99
+    tau: float = 0.95
+    reward_window_size: int = 50
+    use_normalized_advantage: bool = False
+    hidden_size: int = 512
+    entropy_target_factor: float = 0.0
+    use_adaptive_entropy_pen: bool = False
+    use_clipped_value_loss: bool = True
+    # Use double buffered sampling, typically helps
+    # when environment time is similar or larger than
+    # policy inference time during rollout generation
+    # Not that this does not change the memory requirements
+    use_double_buffered_sampler: bool = False
+
+
+@dataclass
+class VERConfig(HabitatBaselinesBaseConfig):
+    """Variable experience rollout config"""
+    variable_experience: bool = True
+    num_inference_workers: int = 2
+    overlap_rollouts_and_learn: bool = False
+
+
+@dataclass
+class AuxLossConfig(HabitatBaselinesBaseConfig):
+    pass
+
+
+@dataclass
+class CPCALossConfig(AuxLossConfig):
+    """Action-conditional contrastive predictive coding loss"""
+    k: int = 20
+    time_subsample: int = 6
+    future_subsample: int = 2
+    loss_scale: float = 0.1
+
+
+@dataclass
+class DDPPOConfig(HabitatBaselinesBaseConfig):
+    """Decentralized distributed proximal policy optimization config"""
+    sync_frac: float = 0.6
+    distrib_backend: str = "GLOO"
+    rnn_type: str = "GRU"
```

Comment:
Strange, I would think the rnn_type and other algorithm specific keys would live in the PPOConfig and only distributed things would be in the DDPPOConfig.

Reply:
I just replicated what was defined in the `default.py` in hydra structured config style. I think the implementation logic here is the same as with VER & PPO. Please read my comments about this in the VER related comments in the PR discussion.